### ISSUE TITLE
`derivable_impls` FN when enum is qualified with `Self`

### ISSUE THREAD

**Issue by profetia** [rust-lang/rust-clippy#15536]
### Summary

`derivable_impls` FN when enum is qualified with `Self` in its impl of `Default`

### Lint Name

derivable_impls

### Reproducer

I tried this code:

```rust
#[derive(Copy, Clone)]
enum Bar {
    A,
    B,
}

impl Default for Bar {
    fn default() -> Self {
        Self::A
    }
}
```

I expected to see this happen:

The lint is triggered for `Bar`

Instead, this happened:

Nothing

### Version

```text

```
